### DISCUSSION TITLE
Build retrofit AAR files from this repo

### DISCUSSION THREAD

**Discussion by ivasku** [square/retrofit#3591] [Q&A]
I have downloaded the source code, and I have managed to open inside Android Studio and the build passes:

![image](https://user-images.githubusercontent.com/5948031/126297260-86f3de27-a53e-4f12-8541-c6906bb243be.png)

But I am not sure how can I get an artifact from this ? There is no output DIR with AAR files that should be build , eg, Retrofit, its converters libs... etc.

Did anyone tried to do this ? What am I missing ?

Thank you

**Comment by cortinico** [comment]
> Did anyone tried to do this ? What am I missing ?

Just a heads up, Retrofit is not an Android library so you won't find `.aar` anywhere. You should look for a `.jar` inside your `build` folder after you called the `assemble` gradle task.

**Reply by ivasku** [reply]
Thanks, I have looked for .jar files also, there is no output DIR, no build DIR anywhere, like I did not built the project.

**Comment by JakeWharton** [comment]
Each module has an `assemble` task which will place a `.jar` in `build/libs/`. So if you wanted just the main artifact you would run `./gradlew :retrofit:assemble` and look in `retrofit/build/libs/`.

**Reply by ivasku** [reply]
I did it. Thank you
